{"version":3,"sources":["../../app-src/util/ApplicationException.js"],"names":["ApplicationException","Error","constructor","msg","name","exception","isApplicationException","err","Object","getPrototypeOf","getExceptionMessage","message","console","log"],"mappings":";;;;;;AAAO,YAAMA,oBAAN,SAAmCC,KAAnC,CAAyC;AAC9CC,oBAAYC,MAAM,EAAlB,EAAsB;AACpB,gBAAMA,GAAN;AACA,eAAKC,IAAL,GAAY,KAAKF,WAAL,CAAiBE,IAA7B;AACD;AAJ6C;;;;AAOhD,YAAMC,YAAYL,oBAAlB;;AAEO,eAASM,sBAAT,CAAgCC,GAAhC,EAAqC;AAC1C,eAAOA,eAAeF,SAAf,IACLG,OAAOC,cAAP,CAAsBF,GAAtB,aAAsCF,SADxC;AAED;;;;AAEM,eAASK,mBAAT,CAA6BH,GAA7B,EAAkC;AACvC,YAAGD,uBAAuBC,GAAvB,CAAH,EAAgC;AAC9B,iBAAOA,IAAII,OAAX;AACD,SAFD,MAEO;AACLC,kBAAQC,GAAR,CAAYN,GAAZ;AACA,iBAAO,uCAAP;AACD;AACF","file":"ApplicationException.js","sourcesContent":["export class ApplicationException extends Error {\r\n  constructor(msg = '') {\r\n    super(msg);\r\n    this.name = this.constructor.name;\r\n  }\r\n}\r\n\r\nconst exception = ApplicationException;\r\n\r\nexport function isApplicationException(err) {\r\n  return err instanceof exception ||\r\n    Object.getPrototypeOf(err) instanceof exception\r\n}\r\n\r\nexport function getExceptionMessage(err) {\r\n  if(isApplicationException(err)) {\r\n    return err.message;\r\n  } else {\r\n    console.log(err);\r\n    return 'Não foi possível realizar a operação.';\r\n  }\r\n}"]}